machine-learning/db/QueryBuilder.py

```diff
@@ -215,13 +218,22 @@ def get_refactoring_levels(dataset="") -> str:
 def __get_level(instance_name: str, level: int, m_refactoring: str, dataset: str = "") -> str:
     refactoring_condition: str = instance_name + ".level = " + str(level)
     if m_refactoring != "":
-        refactoring_condition += " AND " + refactoringCommits + ".refactoring = \"" + m_refactoring + "\""
+        refactoring_condition += " AND " + refactoringCommits + ".refactoring = \"" + m_refactoring + "\""\
+                                 + file_type_filter()
 
     return get_instance_fields(instance_name, [(instance_name, tableMap[instance_name][1]),
                                                (commitMetaData, ["commitDate"])] + get_metrics_level(level),
                                refactoring_condition, dataset, " order by " + commitMetaData + ".commitDate")
 
 
+# Add restriction whether to use only production, test or both files
+def file_type_filter() -> str:
```

Comment:
This class is totally new to me, but the implementation seems clear enough to me!